dsgrid/config/project_config.py

```diff
@@ -166,6 +232,11 @@ class InputDatasetModel(DSGBaseModel):
         updateable=False,
         # TODO: add notes about warnings for outdated versions? DSGRID-189.
     )
+    required_dimensions: RequiredDimensionsModel = Field(
```

Comment:
Something that might be confusing about this implementation is that we use a supplemental dimension record ID to uniquely identify supplemental dimensions while in other places we use a dimension query name. There is a built-in assumption that there is only one record ID for these type of supplemental dimensions. It could be particularly confusing because the record ID is likely to be the same as the dimension query name.

An alternative solution is to have one instance of RequiredDimensionsModel for base dimension records and a second for supplemental dimension query names.

Reply:
Ah, yes, query names do not exist yet, so this idea isn't great. I started off with record IDs because that is what is used for mapping between base and supplemental. As it happens, these supplemental record IDs and query names will be either be identical or very close. We are definitely setting up our users for some confusion.

Reply:
The revision distinguishes between base and supplemental dimensions so the latter can be clearly referenced (thus avoiding name conflict issues).